catalog/spatial/wfs/spatial-wfs-api/src/main/java/org/codice/ddf/spatial/ogc/wfs/catalog/mapper/MetacardMapper.java

```diff
@@ -35,4 +39,31 @@
 
   /** Gets the unit of the feature property mapped the metacard resource-size */
   String getDataUnit();
+
+  /**
+   * @param p - a predicate to identify the entry we're looking for. Passing in 'null' returns
+   *     Optional.empty().
+   * @return a matching entry found for which 'p.apply()' returns 'true'. It's expected that 'p'
+   *     will uniquely identify an entry. When 'p' matches multiple Entry objects, the one selected
+   *     is implementation-dependent.
+   */
+  Optional<Entry> getEntry(@Nullable Predicate<Entry> p);
+
+  /** @return a java.util.Stream of the entries that have been added to this MetacardMapper. */
+  Stream<Entry> stream();
+
+  /** Represents a mapping between metacard attribute name and WFS feature name. */
+  interface Entry {
+    /** @return the name of the WFS feature to be mapped */
+    String getFeatureProperty();
+
+    /** @return the name of the metaacard attribute that equates to the WFS feature */
+    String getAttributeName();
+
+    /**
+     * @return returns a function that, given a map of key/value pairs, returns a value for given
+     *     mapping. The mechanism used to determine that value is implementation dependent.
+     */
+    Function<Map<String, Serializable>, String> getMappingFunction();
```

Comment:
It looks like the implementation in this PR returns null for this method.  We may want to either annotate it with `@Nullable`